packages/actor-init-sparql/test/HttpServiceSparqlEndpoint-test.ts

```diff
@@ -761,3 +686,360 @@ describe('HttpServiceSparqlEndpoint', () => {
+{ type: 'application/sparql-results+json',
+  quality: 1,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 1,
+  q: 0.4 },
+{ type: 'application/sparql-results+xml',
+  quality: 1,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 1,
+  q: 0.4 },
+{ type: 'table',
+  quality: 1,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 1,
+  q: 0.4 },
+{ type: 'application/trig',
+  quality: 1,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 1,
+  q: 0.4 },
+{ type: 'simple',
+  quality: 1,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 1,
+  q: 0.4 },
+{ type: 'tree',
+  quality: 0.9,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 0.9,
+  q: 0.36 },
+{ type: 'application/ld+json',
+  quality: 0.9,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 0.9,
+  q: 0.36 },
+{ type: 'application/n-quads',
+  quality: 0.7,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 0.7,
+  q: 0.28 },
+{ type: 'text/turtle',
+  quality: 0.6,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 0.6,
+  q: 0.24 },
+{ type: 'application/n-triples',
+  quality: 0.3,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 0.3,
+  q: 0.12 },
+{ type: 'text/n3',
+  quality: 0.2,
+  qm: 1,
+  qt: 0.8,
+  qts: 2,
+  ql: 0.5,
+  qls: 0,
+  qc: 1,
+  qcs: 0,
+  qe: 1,
+  qes: 0,
+  qs: 0.2,
+  q: 0.08 }];
+
+describe('HandleNegotiation', () => {
+  let instance: any;
+  let engine: any;
+  const stdout = new PassThrough();
+  const stderr = new PassThrough();
+  let request: any;
+  let response: any;
+  let variants: any;
+
+  beforeEach(async() => {
+    instance = new HttpServiceSparqlEndpoint({});
+    instance.writeQueryResult = jest.fn();
+    engine = await newEngineDynamic();
+    request = makeRequest();
+    response = new ServerResponseMock();
+    variants = [{ type: 'application/trig', quality: 0.4 },
+      { type: 'stats', quality: 1 },
+      { type: 'application/json', quality: 1 },
+      { type: 'simple', quality: 0.5 }];
+  });
+
+  function makeRequest() {
+    request = stringToStream('default_test_request_content');
+    request.url = 'url_sparql';
+    request.headers = { 'content-type': 'contenttypewhichdefinitelydoesnotexist',
+      accept: 'application/json,application/trig,simple,stats' };
+    request.method = 'GET';
+    return request;
+  }
+
+  it('should call writeQueryResult with the preferred default'
+    , async() => {
+      const choose = jest.fn(() => negotiation);
+      jest.doMock('negotiate', () => {
```

Comment:
Is there a need to mock? Testing-wise, it would be safer if we could call the original logic.